src/pages/authenticationPage/Signin.js

```diff
@@ -61,62 +57,102 @@ const Signin = () => {
-            value={setUserInfo.email}
-            onChange={(event) =>
-              setUserInfo({ ...userInfo, email: event.target.value })
+        <Formik
+          className="form"
+          onSubmit={(values) => {
+            loginHandler(values);
+          }}
+          initialValues={{
+            email: "",
+            password: "",
+          }}
+          validate={(values) => {
+            const errors = {};
+            if (!values.password.trim()) {
+              errors.password = "Required";
             }
-          />
-          <label htmlFor="password" className="form-label">
-            Password
-          </label>
-          <div className="show-hide-password">
-            <input
-              id="create-password"
-              className="input"
-              type="password"
-              placeholder="Enter Password"
-              name="password"
-              required
-              value={setUserInfo.password}
-              onChange={(event) =>
-                setUserInfo({ ...userInfo, password: event.target.value })
-              }
-            />
-            <div className="show-password" id="show-create-password">
-              show
+            if (!values.email) {
+              errors.email = "Required";
+            } else if (
+              !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i.test(values.email)
+            ) {
+              errors.email = "Invalid email address";
+            }
+            return errors;
+          }}
+        >
+          <Form className="form">
+            <div className="login-heading">Login</div>
+            <div>
+              <label htmlFor="email" className="form-label">
+                Email
+              </label>
+              <Field
+                className="input"
+                type="text"
+                placeholder="Enter Email"
+                name="email"
+              />
+              <ErrorMessage
+                className="error-message"
+                name="email"
+                component="div"
+              />
             </div>
-          </div>
 
-          <div className="login-store flex flex-space-between">
-            <label htmlFor="store">
-              <input type="checkbox" />I accept all terms & conditions
-            </label>
-          </div>
+            <div>
+              <label htmlFor="password" className="form-label">
+                Password
+              </label>
+              <div className="show-hide-password">
+                <Field
+                  id="create-password"
+                  className="input"
+                  type={showPassword ? "text" : "password"}
+                  placeholder="Enter Password"
+                  name="password"
+                />
+                <div
+                  className="show-password"
+                  onClick={() => {
+                    setShowPassword(!showPassword);
+                  }}
+                >
+                  show
```

Comment:
You can use an icon here instead of text show

Reply:
will do